### PR TITLE
Remove instance checks in PositionAppenders

### PR DESCRIPTION
**Pull request by sopel39**
Current Type contract does not enforce
    particular block type for data.
    Hence PositionAppenders should work with
    generic Block type.

## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

> How would you describe this change to a non-technical end user or system administrator?

## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# General
* Fix query failures when UUID type is used. ({issue}`issuenumber`)
```

Fixes: https://github.com/trinodb/trino/issues/13265

### CODE REVIEW

**Comment by sopel39** [comment]
cc @lukasz-stec

**Review by lukasz-stec — APPROVED** [review]
I would drop the TODOs, multiple `Block` types for a given `Type` are problematic for the entire engine, not only here.
Other than that, lgtm

File: core/trino-main/src/main/java/io/trino/operator/output/RowPositionsAppender.java [hunk 1]
```diff
@@ -66,13 +66,14 @@ private RowPositionsAppender(PositionsAppender[] fieldAppenders, int expectedPos
...
     @Override
+    // TODO: Make PositionsAppender work performant with different block types (https://github.com/trinodb/trino/issues/13267)
     public void append(IntArrayList positions, Block block)
     {
         if (positions.isEmpty()) {
             return;
         }
         ensureCapacity(positions.size());
-        RowBlock sourceRowBlock = (RowBlock) block;
+        AbstractRowBlock sourceRowBlock = (AbstractRowBlock) block;
```
Comment (martint) [inline core/trino-main/src/main/java/io/trino/operator/output/RowPositionsAppender.java:76]: There are no other implementations of AbstractRowBlock aside from RowBlockBuilder, which would never appear here.

**Review by martint — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/operator/output/RowPositionsAppender.java (hunk 1 above)
Comment (sopel39) [reply core/trino-main/src/main/java/io/trino/operator/output/RowPositionsAppender.java:76]: There could be user implementations... It's SPI

**Comment by findepi** [comment]
Should this have a regression test?

**Pull request merged**